### ISSUE TITLE
[New rules request] Empty class and class with two methods, one of which is init

### ISSUE THREAD

**Issue by Rizhiy** [astral-sh/ruff#11763]
Hi, not sure if this is the correct place, please point me in the right direction if not.
I've recently re-watched a [good talk on not writing too many classes](youtube.com/watch?v=o9pEzgHorH0) and was wondering if it is possible to implement a few recommendations giver there as rules.

__Empty class rule__: warn when the class just subclasses another while not doing anything, e.g.:
```python
class MyDict(dict):
	pass
```

__Class with two methods, one of which is `__init__`__: warn when a class has only two methods, one of which is `__init__`, e.g.:
```python
class Greeting:
    def __init__(self, greeting: str = "Hello") -> None:
        self._greeting = greeting

    def greet(self, name) -> None:
        print(f"{self._greeting} {name}!")
```

I did a quick search through existing rules, but couldn't find anything relevant.

**Comment by charliermarsh** [comment]
I think this is roughly what's implemented in https://github.com/astral-sh/ruff/pull/9601. I might suggest chiming in on or watching that PR!
